### PR TITLE
[macOS] Support for FontIcon

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): #2594

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

`FontIcon` is not displayed.

## What is the new behavior?

`FontIcon` displays properly.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/Icon/IconElement.macOS.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+			{
+				Subviews[0].RemoveFromSuperview();
+			}
+		}
+
+		partial void RegisterSubView(NSView child)
+		{
+			if (Subviews.Length != 0)
+			{
+				throw new Exception("A Xaml IconElement may not contain more than one child.");
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Controls/Icon/IconElement.macOS.cs]: Codacy Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4937867)

File: src/Uno.UI/UI/Xaml/Controls/Icon/IconElement.macOS.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿using Windows.UI.Xaml.Controls;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using AppKit;
+
+namespace Windows.UI.Xaml.Controls
+{
+	public partial class IconElement
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/Icon/IconElement.macOS.cs]: Indentation

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/Icon/IconElement.macOS.cs]: Reformated :-)

Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/Controls/Icon/IconElement.macOS.cs]: Still incorrect :-S

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/Icon/IconElement.macOS.cs]: Looks like vs4mac is not handling the .editorconfig file properly. Maybe it's placed at the wrong location.

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/Icon/IconElement.macOS.cs]: Fixed, Parallels' Visual Studio 2019 is more and more useful :-D .

**Comment by MartinZikmund** [comment]
@jeromelaban Had to add one additional commit, as there was a potentially breaking change - using `NSView` instead of `UIElement`, can you please confirm it is still ok?

File: build/PackageDiffIgnore.xml [hunk 3]
```diff
@@ -655,6 +655,9 @@
 				<Member
 					fullName="System.Void Windows.ApplicationModel.DataTransfer.DataPackage.SetText(System.String text)"
 					reason="UWP uses 'value' as the parameter name" />
+				<Member
```
Comment (jeromelaban) [inline build/PackageDiffIgnore.xml]: This method should most likely be internal. The ignore will be useful then :)

File: src/Uno.UI/UI/Xaml/Controls/FontIcon/FontIcon.cs [hunk 4]
```diff
@@ -37,10 +37,10 @@ private void SynchronizeProperties()
 			_textBlock.VerticalAlignment = VerticalAlignment.Center;
 			_textBlock.HorizontalAlignment = HorizontalAlignment.Center;
 			_textBlock.TextAlignment = Windows.UI.Xaml.TextAlignment.Center;
-		}
-
+		}
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/FontIcon/FontIcon.cs]: Ran it through Format Document in full VS, there is actually one additional space in the original file, but not sure why the diff show many "non-changed" lines

File: build/PackageDiffIgnore.xml (hunk 3 above)
Comment (MartinZikmund) [reply build/PackageDiffIgnore.xml]: Changed to internal

File: src/Uno.UI/UI/Xaml/Controls/FontIcon/FontIcon.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/FontIcon/FontIcon.cs]: Probably was a mixed file. Thanks.

File: build/PackageDiffIgnore.xml (hunk 3 above)
Comment (MartinZikmund) [reply build/PackageDiffIgnore.xml]: @jeromelaban Oh, switching to internal means we would need to add all the platform versions into diff ignore - for example `System.Void Windows.UI.Xaml.Controls.IconElement.AddIconElementView(Android.Views.View child)`, etc., is that ok?

Comment (MartinZikmund) [reply build/PackageDiffIgnore.xml]: @jeromelaban I have added the platform variants in the diff ignore, so now the build should go through, also added mention of `AddIconElementView` in breaking changes, can I add ready to merge?

**Pull request merged**